app/helpers/image_encode_helper.rb

```diff
@@ -0,0 +1,33 @@
+module ImageEncodeHelper
+  def encodable_image_tag(source, options = {})
+    image_tag(encodable_image_source(source), options)
+  end
+  alias eimage_tag encodable_image_tag
+
+  def encodable_image_source(source)
+    if base64_encode_images? && source.present?
+      base64_encoded_uri(source)
+    else
+      path_to_image(source)
+    end
+  end
+  alias eimage_source encodable_image_source
+
+  def base64_encode_images?
+    @base64_encode_images
+  end
+
+  def base64_encoded_uri(source)
+    asset = Rails.application.assets[source]
+
+    if asset.content_type == 'image/svg+xml'
+      encoding = 'charset=utf-8'
+      data = CGI.escape(asset.source).gsub('+', '%20')
+    else
+      encoding = 'base64'
+      data = Base64.encode64(asset.source)
+    end
+
+    "data:#{asset.content_type};#{encoding},#{data}"
```

Comment:
Yeah, `Base64` encodes data into data URIs 😜

Seriously though, `Sprockets` does some of this, but it seems difficult to use outside of a stylesheet. You can compare [this method](https://github.com/rails/sprockets/blob/master/lib/sprockets/context.rb#L199) with what I have. And why would we want another dependency?